src/shogun/io/TBOutputFormat.cpp

```diff
@@ -112,7 +112,7 @@ tensorflow::Event TBOutputFormat::convert_scalar(
 		    any_cast<decltype(type)>(value.first.get_value()));
 	};
 
-	sg_for_each_type(value.first.get_value(), sg_all_types, write_summary);
+	sg_for_each_scalar_type(value.first.get_value(), sg_all_types, write_summary);
```

Comment:
mmh this name is a bit of a monster now ;)
I am thinking loud ... sg_dispatch_any, sg_dispatch_any_vector, sg_dispatch_any_matrix, sg_dispatch_any_stl
Don't know if that is even better

Reply:
Including "any" in the function name might make it easier to find the function in docs! So something along those lines would be better.